File: Data Structures & Algorithms/two-integer-sum/submission-1.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap <Integer, Integer> hashmap = new HashMap<>();
        
        int[] solution = new int [2];

        for(int i=0; i<nums.length; i++){
            if(hashmap.containsKey(target-nums[i])){
                solution[0] = hashmap.get(target-nums[i]);
                solution[1] = i;                
            } else {
                hashmap.put(nums[i],i);
            }
        }
        return solution;
    }
}

File: Data Structures & Algorithms/binary-search/submission-1.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        if len(nums) == 0:
            return -1

        mid = len(nums) // 2
        l , r = 0, len(nums) - 1
        while l <= r:
            if nums[mid] == target:
                return mid
            elif nums[mid] > target:
                r = mid - 1
                mid = (l+r) // 2
            else:
                l = mid + 1
                mid = (l+r) // 2
        return -1

        